runtime/src/accounts_index.rs

```diff
@@ -936,13 +944,18 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
         pubkey: &Pubkey,
         ancestors: Option<&Ancestors>,
         max_root: Option<Slot>,
-    ) -> Option<(ReadAccountMapEntry<T>, usize)> {
-        self.get_account_read_entry(pubkey)
-            .and_then(|locked_entry| {
-                let found_index =
-                    self.latest_slot(ancestors, &locked_entry.slot_list(), max_root)?;
-                Some((locked_entry, found_index))
-            })
+    ) -> AccountIndexGetResult<T> {
```

Comment:
Here's the api change where we return an enum instead of an Option.

Reply:
Note that we want 3 states:
1. found in index and matching a root/ancestor (previously was Some(...)
2. found in index, NOT matching a root/ancestor (previously was None)
3. not found in index, (previously was None)

Reply:
Yeah an enum for multiple options makes the most sense to me 😃